File: src-examples/ArrayList_String.java
import java.util.ArrayList;

public class ArrayList_String {

    public static void main(String[] args) {
        ArrayList<String> list = new ArrayList<String>(10);

        list.add(0, new String("C"));
        list.add(1, new String("C#"));
        list.add(2, new String("C++"));
        list.add(3, new String("Java"));
        list.add(4, new String("Perl"));
        list.add(5, new String("Python"));
        list.add(6, new String("Ruby"));

        int i;
        for (i = 0; i < list.size(); i++) {
            String lang = list.get(i);
            System.out.printf("%s\n", lang);
        }

        return;
    }

}

File: src-examples/arraylist_string.c
#include "lang.String.h"
#include "lang.System.h"
#include "util.ArrayList.h"

int main() {
    System_t *sys = System();
    ArrayList_t *list = ArrayList(10);

    list->add(list, 0, String("C"));
    list->add(list, 1, String("C#"));
    list->add(list, 2, String("C++"));
    list->add(list, 3, String("Java"));
    list->add(list, 4, String("Perl"));
    list->add(list, 5, String("Python"));
    list->add(list, 6, String("Ruby"));

    int i;
    for (i = 0; i < list->size(list); i++) {
        String_t *lang = list->get(list, i);
        sys->out->println(lang);
    }

    return 0;
}
